File: src/mb/mizinkobusters/kitpvp/kit/Standard.java
package mb.mizinkobusters.kitpvp.kit;

import mb.mizinkobusters.kitpvp.utils.KitPvPUtils;
import org.bukkit.Material;
import org.bukkit.attribute.Attribute;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerItemConsumeEvent;
import org.bukkit.event.player.PlayerRespawnEvent;
import org.bukkit.inventory.ItemStack;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

public class Standard implements Listener {

    @EventHandler
    public void onKill(PlayerRespawnEvent event) {
        Player player = event.getPlayer();
        if (player.getKiller() == null) {
            return;
        }

        Player killer = player.getKiller();
        if (!KitPvPUtils.isInWorld(killer)) {
            return;
        }

        if (!KitPvPUtils.getKit(killer).equals("Standard")) {
            return;
        }
        killer.getInventory().addItem(new ItemStack(Material.GOLDEN_APPLE));
        killer.setHealth(killer.getAttribute(Attribute.GENERIC_MAX_HEALTH).getValue());
    }

    @EventHandler
    public void onEatGapple(PlayerItemConsumeEvent event) {
        Player player = event.getPlayer();
        if (!KitPvPUtils.isInWorld(player)) {
            return;
        }
        if (!KitPvPUtils.getKit(player).equals("Standard")) {
            return;
        }

        ItemStack item = event.getItem();
        if (!item.getType().equals(Material.GOLDEN_APPLE)) {
            return;
        }

        if (player.hasPotionEffect(PotionEffectType.SPEED)) {
            player.removePotionEffect(PotionEffectType.SPEED);
        }
        player.addPotionEffect(new PotionEffect(PotionEffectType.SPEED, 600, 0));
        player.sendMessage("§7Skill Trigger: §6Standard");
    }
}

File: src/main/java/mb/mizinkobusters/kitpvp/kit/Standard.kt
package mb.mizinkobusters.kitpvp.kit

import org.bukkit.Material
import org.bukkit.event.player.PlayerItemConsumeEvent
import org.bukkit.inventory.ItemStack
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType

object Standard : BaseKit {
    override val lore = listOf(
        "§f頭: 鉄",
        "§f胴: ダイヤ",
        "§f腰: 鉄",
        "§f脚: 鉄",
        "§f武器: 鉄の剣",
        "§f補助: 金のリンゴ×1",
        "§f特殊: 金のリンゴを食べると30秒間移動速度上昇Iが付与される"
    )

    override val displayItemStack = ItemStack(Material.IRON_SWORD)
    override val helmet = ItemStack(Material.IRON_HELMET)
    override val chestplate = ItemStack(Material.DIAMOND_CHESTPLATE)
    override val leggings = ItemStack(Material.IRON_LEGGINGS)
    override val boots = ItemStack(Material.IRON_BOOTS)
    override val weapons = listOf(
        ItemStack(Material.IRON_SWORD),
        ItemStack(Material.GOLDEN_APPLE)
    )

    override fun onEatGapple(event: PlayerItemConsumeEvent) {
        super.onEatGapple(event)

        val player = event.player
        if (player.hasPotionEffect(PotionEffectType.SPEED)) {
            player.removePotionEffect(PotionEffectType.SPEED)
        }
        player.addPotionEffect(PotionEffect(PotionEffectType.SPEED, 600, 0))
        player.sendMessage("§7Skill Trigger: §6Standard")
    }
}